### PR TITLE
Complete the add, sub, and mul operators to complete unit testing and…

### PR DESCRIPTION
**Pull request by wynneyin**

### CODE REVIEW

File: torch/nn/functional.py
Comment (zlyber) [inline torch/nn/functional.py:1]: It is recommended to remove all codes and files related to ```ntt``` module from your branch.

File: torch/testing/_internal/zkp_test_lists.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                    #################################################################
+                base1=self.compute_base_mod(t1,self.modlist[type_element])
+                base2=self.compute_base_mod(t2,self.modlist[type_element])
+
+                r1=F.to_mont(t1)
+                r2=F.to_mont(t2)
+
+                mont1=self.compute_mont(r1)
+                mont2=self.compute_mont(r2)
+                # print(mont1,mont2)
```
Comment (zlyber) [inline torch/testing/_internal/zkp_test_lists.py:142]: still redundant codes.

File: 2.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        output = F.to_mod(in_1)
+        return output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        input, = ctx.saved_tensors
+        grad_input =grad_output
+        return grad_input   
+
+# 构建模型
```
Comment (zlyber) [inline 2.py:46]: no Chinese please...

**Comment by wynneyin** [comment]
Sorry for forgetting this file, I will modify it&nbsp;


尹文浩
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhiyuan ***@***.***&gt;; 
发送时间: 2023年12月13日(星期三) 下午4:39
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PNP-team/PNP] Complete the add, sub, and mul operators to complete unit testing and… (PR #7)





 
@zlyber commented on this pull request.
 
 
In 2.py:
 &gt; +        return grad_input + +class to_mod(torch.autograd.Function): +    @staticmethod +    def forward(ctx, in_1): +        ctx.save_for_backward(in_1) +        output = F.to_mod(in_1) +        return output + +    @staticmethod +    def backward(ctx, grad_output): +        input, = ctx.saved_tensors +        grad_input =grad_output +        return grad_input    + +# 构建模型  
no Chinese please...

**Review by zlyber — CHANGES_REQUESTED** [review]
1. There is still redundant codes and comments, check all your commits more carefully.
2. No Chinese.
3. Only two curves are tested now, should we support other curves at this stage? @jizhuoran

**Comment by wynneyin** [comment]
1. Native_ Function_ Generation.py was modified because I added an equal_ The mod function, the return value of this function is of type bool, not of type tensor.
2. Unittest_ Mod. py is the test file, and I will move it to the test folder

&nbsp;


尹文浩
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhuoran ***@***.***&gt;; 
发送时间: 2023年12月15日(星期五) 下午5:26
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PNP-team/PNP] Complete the add, sub, and mul operators to complete unit testing and… (PR #7)





 
Some quick comments.
 
First, test file like the 1.py, 2.py files should be removed.
 
Second, ntt family function should be removed from functional.py
 
Third, pay attention to the space/new line/{, please follow the original format. You can check the modification in this tab.
 Screenshot.2023-12-15.at.17.22.13.png (view on web)
 
Forth, the function name should follow the same format. For add, it is add_cpu (I think it should be add_mod_cpu), while for sub_mod_, it becomes subtraction_cpu_. Meanwhile, why there is comments like ##div in the yarm file.
 
Fifth, why torchgen/native_function_generation.py is changed?
 
Sixth, what's the function of unittest_mod.py?

**Pull request closed without merging**